packages/framework-provider-local/src/library/graphql-adapter.ts

```diff
@@ -1,45 +1,116 @@
+  requestID: UUID
+): GraphQLRequestEnvelope | GraphQLRequestEnvelopeError {
+  const logger = getLogger(config, 'graphql-adapter#expressWebSocketMessageToEnvelope')
+  logger.debug('Received  WebSocket GraphQL request: ', webSocketRequest)
+
+  let eventType: EventType = 'MESSAGE'
+  const incomingMessage = webSocketRequest.incomingMessage
+  const headers = incomingMessage?.headers
+  const data = webSocketRequest.data as GraphQLOperation | GraphQLClientMessage | undefined
   try {
+    const connectionContext = webSocketRequest.connectionContext
+    eventType = connectionContext?.eventType
     return {
       requestID,
       eventType,
-      connectionID,
-      token: request.headers.authorization,
-      value: request.body,
+      connectionID: connectionContext?.connectionId.toString(),
+      token: headers?.authorization,
+      value: data,
       context: {
         request: {
-          headers: request.headers,
-          body: request.body,
+          headers: headers,
+          body: data,
         },
-        rawContext: request,
+        rawContext: webSocketRequest,
       },
     }
   } catch (e) {
-    const error = e as Error
     return {
-      error,
-      requestID,
-      connectionID,
-      eventType,
+      error: e,
+      requestID: requestID,
+      connectionID: undefined,
+      eventType: eventType,
+      context: {
+        request: {
+          headers: headers,
+          body: data,
+        },
+        rawContext: webSocketRequest,
+      },
+    } as GraphQLRequestEnvelopeError
+  }
+}
+
+function expressHttpMessageToEnvelope(
+  config: BoosterConfig,
+  httpRequest: express.Request,
+  requestId: UUID
+): GraphQLRequestEnvelope | GraphQLRequestEnvelopeError {
+  const logger = getLogger(config, 'graphql-adapter#expressHttpMessageToEnvelope')
+  const eventType: EventType = 'MESSAGE'
+  const headers = httpRequest.headers
+  const data = httpRequest.body
+  try {
+    logger.debug('Received GraphQL request: \n- Headers: ', headers, '\n- Body: ', data)
+    return {
+      connectionID: undefined,
+      requestID: requestId,
+      eventType: eventType,
+      token: headers?.authorization,
+      value: data,
       context: {
         request: {
-          headers: request.headers,
-          body: request.body,
+          headers: headers,
+          body: data,
         },
-        rawContext: request,
+        rawContext: httpRequest,
       },
     }
+  } catch (e) {
+    return {
+      error: e,
+      requestID: requestId,
+      connectionID: undefined,
+      eventType: eventType,
+      context: {
+        request: {
+          headers: headers,
+          body: data,
+        },
+        rawContext: httpRequest,
+      },
+    } as GraphQLRequestEnvelopeError
   }
 }
+
+function isExpressWebSocketMessage(
+  request: express.Request | ExpressWebSocketMessage
+): request is ExpressWebSocketMessage {
+  return Boolean((request as ExpressWebSocketMessage).connectionContext)
```

Comment:
You can use the `in` operator to simplify this:
```suggestion
  return "connectionContext" in request
```